### ISSUE TITLE
React and react-dom should be peerDependencies

### ISSUE THREAD

**Issue by simonedavico** [react-querybuilder/react-querybuilder#80]
Since this component will be consumed from React applications, I believe you can safely assume that both `react` and `react-dom` will be installed alongside `react-querybuilder`.

Right now, `react` is a dependency (the same holds for `react-dom`):

```
•100% ➜ npm ls react
@myscope/myapp /path/to/my/app
├── react@16.8.1
└─┬ react-querybuilder@1.4.3
  └── react@16.8.1  deduped
```

I believe that `react` and `react-dom` should be `peerDependencies`, and `devDependencies` for development.

**Comment by miphe** [comment]
I agree, react and react-dom is only required when running the demo or running tests.

**Comment by jakeboone02** [comment]
This can be closed, right?  `react` and `react-dom` are listed as `peerDependencies` in https://github.com/sapientglobalmarkets/react-querybuilder/blob/master/package.json

**Comment by simonedavico** [comment]
They are still listed also as dependencies though:

https://github.com/sapientglobalmarkets/react-querybuilder/blob/68c266831062f7269c2867c065a3dbd3fcc7fcc9/package.json#L67

https://github.com/sapientglobalmarkets/react-querybuilder/blob/68c266831062f7269c2867c065a3dbd3fcc7fcc9/package.json#L68

Unless I'm wrong, they should be `devDependencies` for component development purposes, and `peerDependencies` for consumers of the component.

**Comment by jakeboone02** [comment]
OK, I [made the change](https://github.com/sapientglobalmarkets/react-querybuilder/commit/dd6ddd21644bda6cd6be765f2ec91cd5382e0ced).  I also fixed the `peerDependencies` to require `">=16.8.0"` instead of `"^16.8.0"`.
